src/pe/delay_import.rs

```diff
@@ -0,0 +1,529 @@
+            .gread_with::<DelayImportThunk>(&mut self.offset, ctx)
+            .ok()?;
+        if thunk.is_null() {
+            self.bytes = &[];
+            return None;
+        }
+
+        let ctx = (self.descriptor, self.ctx);
+        let func = self.bytes.pread_with::<DelayImportFunction>(0, ctx).ok()?;
+
+        Some(Ok(func))
+    }
+}
+
+/// Represents a single entry in the delay import table of a PE (Portable Executable) file.
+#[derive(Debug)]
+pub struct DelayImportEntry<'a> {
+    /// The associated delay import descriptor.
+    pub descriptor: DelayImportDescriptor,
+    /// Name offset of this delay import entry.
+    pub offset: u32,
+    /// RVA of this delay import entry.
+    pub rva: u32,
+    /// Hint value for symbol lookup.
+    pub hint: u16,
+    /// Name of the DLL this function is imported from.
+    pub dll: &'a str,
+    /// Name of the imported function, if available.
+    ///
+    /// This is `None` if the import is by ordinal.
+    pub name: Option<&'a str>,
+    /// Ordinal number if the function is imported by ordinal.
+    ///
+    /// This is `0` if `name` is `Some` and vice versa.
+    pub ordinal: u16,
+}
+
+/// Internal helper struct to simplify bitflag operations.
+#[derive(Debug, SizeWith)]
+struct DelayImportThunk(pub u64);
+
+impl<'a> ctx::TryFromCtx<'a, Ctx> for DelayImportThunk {
+    type Error = crate::error::Error;
+    fn try_from_ctx(bytes: &'a [u8], ctx: Ctx) -> error::Result<(Self, usize)> {
+        let mut offset = 0;
+
+        let value = if ctx.is_big() {
+            bytes.gread_with::<u64>(&mut offset, scroll::LE)?
+        } else {
+            let v = bytes.gread_with::<u32>(&mut offset, scroll::LE)?;
+            v as u64
+        };
+
+        Ok((Self(value), offset))
+    }
+}
+
+impl DelayImportThunk {
+    /// Whether the entry is imported by ordinal (64-bit)
+    pub fn is_ordinal64(&self) -> bool {
+        self.0 & IMPORT_BY_ORDINAL_64 != 0
+    }
+
+    /// Whether the entry is imported by ordinal (32-bit)
+    pub fn is_ordinal32(&self) -> bool {
+        (self.0 as u32) & IMPORT_BY_ORDINAL_32 != 0
+    }
+
+    /// Returns the ordinal value
+    ///
+    /// Use this function if either `is_ordinal64` or `is_ordinal32` returns `true`.
+    pub fn ordinal(&self) -> u16 {
+        (self.0 as u16) & 0xFFFFu16
+    }
+
+    /// Returns the name rva value
+    ///
+    /// Use this function if either `is_ordinal64` or `is_ordinal32` returns `false`.
+    pub fn name_rva(&self) -> u32 {
+        self.0 as u32
+    }
+
+    /// Checks if the underlying value is zero.
+    pub fn is_null(&self) -> bool {
+        self.0 == 0
+    }
+}
+
+/// Represents a delay-load import descriptor in a PE (Portable Executable) file.
+///
+/// This structure corresponds to the `IMAGE_DELAYLOAD_DESCRIPTOR` defined in the PE/COFF specification.
+/// Each instance describes a single delay-loaded DLL and its associated import data.
+#[derive(Debug, Copy, Clone, Pread, Pwrite, SizeWith)]
```

Comment:
just curious why you're deriving SizeWith here and above; this is for the IOwrite traits, which need a size given a ctx so that it can write into a stack buffer.

Reply:
This is just to follow as all others, I think it's a traditional way in goblin to make them pwrite/sizewith? otherwise at least they aren't needed for the parser to function.